projects/cashmere/src/lib/datepicker/calendar/calendar.component.ts

```diff
@@ -532,7 +532,14 @@ export class CalendarComponent implements AfterContentInit, AfterViewChecked, On
             curHour++;
             const tempCycle = this._hourCycle > 12 ? 23 : 12;
             if (curHour > tempCycle) {
-                curHour = tempCycle;
+                curHour = this._hourCycle > 12 ? 0 : 1;
```

Comment:
I don't really understand what's being accomplished here (I don't really know the code for the date picker stuff)

Reply:
@jonest This line allows the 12 hour or 24 hour clock to cycle instead of having a max of 12 and 24 hours respectively and a min of 1 and 0 hours respectively.

Reply:
It just allows the time to reset at its lowest or highest hours instead of remaining at its lowest or highest hours